### PR TITLE
PAAS-1767: apply datadog apm settings for jahia process only when APM is

### PR DESCRIPTION
**Pull request by Ronan-Gorain**
enabled

JIRA issue: https://jira.jahia.org/browse/PAAS-1767

Short description:

### CODE REVIEW

File: packages/jahia/migrations/migrate-to-v9.yaml [hunk 1]
```diff
@@ -37,12 +37,13 @@ onInstall:
   - updateProxySQLConfigForSingleDBNodes           # PAAS-1647
   - addVaultInfosToDBNodes                         # PAAS-1722
   - addJvmRoute                                    # PAAS-1640
+  - updateApmJavaOption                            # PAAS-1767
```
Comment (kevinvandel) [inline packages/jahia/migrations/migrate-to-v9.yaml:40]: If only we had a specific block for all actions that require a restart... Oh wait, it's just below ! :trollface:

**Review by kevinvandel — CHANGES_REQUESTED** [review]

**Review by kevinvandel — DISMISSED** [review]

File: mixins/jahia.yml [hunk 2]
```diff
@@ -181,7 +181,9 @@ actions:
         grep -q '^JAVA_OPTS.*-DmaxPostSize=' ${tomcat_env} || sed -i -E '/^JAVA_OPTS/ s/(.)$/ -DmaxPostSize=${tomcat_cfg_maxpostsize}\1/g' ${tomcat_env}
         grep -q '^JAVA_OPTS.*-XX:NativeMemoryTracking=' ${tomcat_env} || sed -i -E '/^JAVA_OPTS/ s/(.)$/ -XX:NativeMemoryTracking=summary\1/g' ${tomcat_env}
         # Datadog APM
-        sed -i -E '/^JAVA_OPTS/ s#(.)$# -Ddd.profiling.enabled=true -XX:FlightRecorderOptions=stackdepth=256 -Ddd.logs.injection=true -javaagent:/opt/tomcat/datadog/dd-java-agent.jar -Ddd.service=jahia -Ddd.env=${env.domain}\1#g' ${tomcat_env}
+        sed -i '/^JVM_SUFFIX/i if $DATADOG_APM_ENABLED; then\n    APM_OPTS="-Ddd.profiling.enabled=true -XX:FlightRecorderOptions=stackdepth=256 -Ddd.logs.injection=true -javaagent:\/opt\/tomcat\/datadog\/dd-java-agent.jar -Ddd.service=jahia -Ddd.env=${env.domain}"\nfi' $tomcat_env
+        sed -i 's/^\(JAVA_OPTS.*\)"$/\1 $APM_OPTS"/g' /opt/tomcat/conf/tomcat-env.sh
```
Comment (laurentfufu) [inline mixins/jahia.yml:185]:
```suggestion
        echo 'APM_OPTS="-Ddd.profiling.enabled=true -XX:FlightRecorderOptions=stackdepth=256 -Ddd.logs.injection=true -javaagent:/opt/tomcat/datadog/dd-java-agent.jar -Ddd.service=jahia -Ddd.env=${env.domain}"' >> $tomcat_env
        echo '$DATADOG_APM_ENABLED && JAVA_OPTS="$JAVA_OPTS $APM_OPTS"' >> $tomcat_env
```

suggestion for readability

File: packages/jahia/migrations/migrate-to-v9.yaml [hunk 3]
```diff
@@ -132,6 +133,16 @@ actions:
...
         /opt/tomcat/conf/server.xml
 
+  updateApmJavaOption:
+    cmd[proc, cp]: |-
+      # Remove all options set after dd.profiling.enable (basically all used for apm)
+      sed -i 's/-Ddd.profiling.enable.*"/"/g' /opt/tomcat/conf/tomcat-env.sh
+      if [ "$(grep APM_OPTS /opt/tomcat/conf/tomcat-env.sh)" == "" ];then
+        # add apm options in an if block depending on DATADOG_APM_ENABLED value
+        sed -i '/^JVM_SUFFIX/i if $DATADOG_APM_ENABLED; then\n    APM_OPTS="-Ddd.profiling.enabled=true -XX:FlightRecorderOptions=stackdepth=256 -Ddd.logs.injection=true -javaagent:\/opt\/tomcat\/datadog\/dd-java-agent.jar -Ddd.service=jahia -Ddd.env=${env.domain}"\nfi' /opt/tomcat/conf/tomcat-env.sh
+        sed -i 's/^\(JAVA_OPTS.*\)"$/\1 $APM_OPTS"/g' /opt/tomcat/conf/tomcat-env.sh
```
Comment (laurentfufu) [inline packages/jahia/migrations/migrate-to-v9.yaml:143]:
```suggestion
        echo 'APM_OPTS="-Ddd.profiling.enabled=true -XX:FlightRecorderOptions=stackdepth=256 -Ddd.logs.injection=true -javaagent:/opt/tomcat/datadog/dd-java-agent.jar -Ddd.service=jahia -Ddd.env=${env.domain}"' >> $tomcat_env
        echo '$DATADOG_APM_ENABLED && JAVA_OPTS="$JAVA_OPTS $APM_OPTS"' >> $tomcat_env
```
```suggestion
        echo 'APM_OPTS="-Ddd.profiling.enabled=true -XX:FlightRecorderOptions=stackdepth=256 -Ddd.logs.injection=true -javaagent:/opt/tomcat/datadog/dd-java-agent.jar -Ddd.service=jahia -Ddd.env=${env.domain}"' >> $tomcat_env
        echo '$DATADOG_APM_ENABLED && JAVA_OPTS="$JAVA_OPTS $APM_OPTS"' >> $tomcat_env
```

File: packages/jahia/migrations/migrate-to-v9.yaml [hunk 4]
```diff
@@ -132,6 +133,16 @@ actions:
         -i "Server/Service/Engine[not(@jvmRoute)]" -t attr -n "jvmRoute" -v "$short_name" \
         /opt/tomcat/conf/server.xml
 
+  updateApmJavaOption:
+    cmd[proc, cp]: |-
+      # Remove all options set after dd.profiling.enable (basically all used for apm)
+      sed -i 's/-Ddd.profiling.enable.*"/"/g' /opt/tomcat/conf/tomcat-env.sh
+      if [ "$(grep APM_OPTS /opt/tomcat/conf/tomcat-env.sh)" == "" ];then
```
Comment (laurentfufu) [inline packages/jahia/migrations/migrate-to-v9.yaml:140]:
```suggestion
      if ! (grep -q APM_OPTS /opt/tomcat/conf/tomcat-env.sh);then
```

File: mixins/jahia.yml (hunk 2 above)
Comment (laurentfufu) [reply mixins/jahia.yml:185]: `JAVA_OPTS="$JAVA_OPTS $APM_OPTS"` can also be `JAVA_OPTS+=" $APM_OPTS"`

Comment (Ronan-Gorain) [reply mixins/jahia.yml]: Improvements done

File: packages/jahia/migrations/migrate-to-v9.yaml [hunk 5]
```diff
@@ -132,6 +133,16 @@ actions:
...
         /opt/tomcat/conf/server.xml
 
+  updateApmJavaOption:
+    cmd[proc, cp]: |-
+      # Remove all options set after dd.profiling.enable (basically all used for apm)
+      sed -i 's/\s-Ddd\.[^"]*//' /opt/tomcat/conf/tomcat-env.sh
+      if ! (grep -q APM_OPTS /opt/tomcat/conf/tomcat-env.sh);then
+          echo 'APM_OPTS="-Ddd.profiling.enabled=true -XX:FlightRecorderOptions=stackdepth=256 -Ddd.logs.injection=true -javaagent:/opt/tomcat/datadog/dd-java-agent.jar -Ddd.service=jahia -Ddd.env=${env.domain}"' >>  /opt/tomcat/conf/tomcat-env.sh
+          echo '$DATADOG_APM_ENABLED && JAVA_OPTS+="$APM_OPTS"' >> /opt/tomcat/conf/tomcat-env.sh
+
```
Comment (laurentfufu) [inline packages/jahia/migrations/migrate-to-v9.yaml:143]: image

:trollface:

Comment (Ronan-Gorain) [reply packages/jahia/migrations/migrate-to-v9.yaml:143]: fixed

**Review by laurentfufu — APPROVED** [review]

**Pull request merged**